### DISCUSSION TITLE
How to start a new multiplexer domain?

### DISCUSSION THREAD

**Discussion by mhanberg** [wezterm/wezterm#1318] [Q&A]
Hi!

I am trying to see if i can replace tmux with just Wezterm, and I am a little confused on how the wezterm multiplexing is supposed to work.

From what I can tell, a "domain" is similar to a tmux "session", in that it is a collection of tabs (in tmux, a tab is called a "window") and panes.

This thing I'm sort of stuck on is how to create a new domain on the fly and how to switch between the the domains. I'm currently under the impression that I can get the GUI to disconnect from one domain and reconnect to another, similar to how a tmux client can switch which session it's connected to.

This is what I have in my config right now.

```lua
  unix_domains = {
    {
      name = "default",
      connect_automatically = true,
    },
  },
```

I am hoping to recreate the following workflow.

I can hit `prefix ctrl-j` and a floating window with a fzf picker loaded up with the current tmux sessions, and when i pick one from there, the client will switch to that session.

I can also hit `prefix ctrl-y` to do a similar thing, except the fzf window will have a list of all the directories in `~/Development`, and picking one will create a new tmux session in that dir and the client will switch to it.

Note: this would all be from a single GUI window.

**Comment by wez** [comment]
We don't currently have a way to do this dynamically, but you may want to take a look at this post where @kitten has created a helper script to make this sort of functionality: https://github.com/wez/wezterm/discussions/1322
